### PR TITLE
Add more complex docs for Contessa

### LINKED ISSUE #1: Add documentation
Add proper documentation for purpose and basic usage.

### PR DESCRIPTION
**Pull request by maroshmka**
closes #1

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -6,7 +6,9 @@ Hello, welcome to Contessa documentation!
 Quick Example
```
Comment (junckritter) [inline docs/index.rst]: I miss some general description of the project, for user to understand the purpose at first glance. There is some info in `quick_start`, but not very much.

Comment (maroshmka) [reply docs/index.rst]: Yep, thats true, gonna add some more to readme and link it in index :)

File: docs/features.rst [hunk 2]
```diff
@@ -25,37 +23,40 @@ Format
...
 
 .. code-block:: json
 
     {
         "time_filter": "a"
     }
 
-- Parallel to **columns** shortcut for defining multiple rules at once, parameter **separate_time_filters** is available.
-- Each one will be used separately. In case both *separate_time_filters* and *time_filter* are defined, only *separate_time_filters* would be considered.
+- Parameter **separate_time_filters** is available.
```
Comment (m1n0) [inline docs/features.rst]: **separate_time_filters** parameter

File: docs/features.rst [hunk 3]
```diff
@@ -25,37 +23,40 @@ Format
...
+
+  - Each one will be used separately. Meaning, it would actually create X rules and therefore X checks would be done against the data, each filtering the data by passed column. In case both *separate_time_filters* and *time_filter* are defined, only *separate_time_filters* would be considered.
```
Comment (m1n0) [inline docs/features.rst]: This seems to be unclear, did I get it correctly? `Creates a matrix of rules and filters - i.e. provided rule is applied with each separate_time_filters value separately - acts as a generator of rules.`

File: docs/features.rst [hunk 4]
```diff
@@ -25,37 +23,40 @@ Format
...
 
 .. code-block:: json
 
     {
+        "name": NOT_NULL,
+        "columns": ["a", "b", "c"],
         "separate_time_filters": [{"column": "c"}, {"column": "d"}]
     }
+    # would check NOT_NULL while filtering by "c" column for last 30 days, write the result.
```
Comment (m1n0) [inline docs/features.rst]: Checks `NOT_NULL` while filtering by column `c` and writes the result, then repeats the same check while filtering by column `d` and writes the result as a separate value.

File: docs/features.rst [hunk 5]
```diff
@@ -65,8 +66,19 @@ Each run has its own context, mostly used for templating the final sql. This is
...
 
-Postgres result table
+Quality Check Result
 -------------------------
 
-# todo
\ No newline at end of file
+.. quality-check-start
+
+Result will be storred to ``result_table`` in your database. It's model is defined in :ref:`quality_check`.
```
Comment (m1n0) [inline docs/features.rst]: "It`s" means "It is", so correct form is "Its"

File: docs/quick_start.rst [hunk 6]
```diff
@@ -1,4 +1,131 @@
 How to write your first check
 ==============================
 
-# todo
\ No newline at end of file
+Basic idea of Contessa is to help you with defining data quality rules
```
Comment (m1n0) [inline docs/quick_start.rst]: I would get rid of "you"s, i.e. `Basic idea of Contessa is to help with... ` then `Let's create a script ` -> `Create a script... `

File: docs/quick_start.rst [hunk 7]
```diff
@@ -1,4 +1,131 @@
...
+	    },
+	    {
+	        "name": GT,
+	        "value": 0,
+	        "columns": ["full_price", "initial_price"],
+	        "condition": "status IN ('closed', 'refunded')",
+    	},
+	]
+
+Let's analyze what's happenning here:
```
Comment (m1n0) [inline docs/quick_start.rst]: `Explanation:`

File: docs/quick_start.rst [hunk 8]
```diff
@@ -1,4 +1,131 @@
...
+	      cur.execute("select column_name from information_schema.columns where table_name = 'abc' and data_type = 'integer';")
+	      columns = [c[0] for c in cur.fetchall()]
+	  return columns
+
+	RULES.append({
+		"name": NOT_NULL,
+		"columns": get_cool_cols()
+	})
+
+Which will create rule that checks if all integer columns in table 'abc' are not null. Of course, you can go much wilder, get all tables you have, get all columns in those tables and for each one crates a rule. You could even spawn multiple runners for each dbs you have or want to check.
```
Comment (m1n0) [inline docs/quick_start.rst]: create a rule

File: docs/quick_start.rst [hunk 9]
```diff
@@ -1,4 +1,131 @@
...
+
+	from contessa import ContessaRunner
+	contessa = ContessaRunner("postgres://:@localhost:5432")
+	contessa.run(
+        raw_rules=RULES,
+        check_table={"schema_name": "public", "table_name": "bookings"},
+        result_table={"schema_name": "dq", "table_name": "my_table"},
+    )
+
+Yop, thats it. Runner goes against 1 db, take your rules, do a bit wrangling with normalization you do not
```
Comment (m1n0) [inline docs/quick_start.rst]: I think this should be `Runner runs` ?

Comment (m1n0) [inline docs/quick_start.rst]: takes your rules, does a bit of wrangling...~~it~~ creates...

File: docs/quick_start.rst [hunk 10]
```diff
@@ -1,4 +1,131 @@
...
+--------------------------------------
+
+.. include:: features.rst
+  :start-after: quality-check-start
+  :end-before: quality-check-end
+
+Where to look next
+--------------------------------------
+
+Go ahead and test it all with your local db! Then you can schedule a cron to have this results regularly :)
```
Comment (m1n0) [inline docs/quick_start.rst]: `...cron to run these checks regularly ` or `...to get these results regularly`

File: docs/quick_start.rst [hunk 11]
```diff
@@ -1,4 +1,131 @@
...
+
+If you wanna use Contessa even more, you can look at detailed description of:
```
Comment (m1n0) [inline docs/quick_start.rst]: ~~wanna~~ want to

File: docs/rules.rst [hunk 12]
```diff
@@ -1,37 +1,82 @@
...
+    "sql": "SELECT CASE WHEN <something-cool> THEN false ELSE true END FROM {{table_fullname}}",
+    "description": "more coolness",
+    # condition doesnt make sense here :)
+  }]
+
+.. tip::
+
+	value can be another col, e.g. - ``{"name": NOT, "column": "src", "value": "dst"}``
+
+Below you can check what rules we are supporting.
```
Comment (m1n0) [inline docs/rules.rst]: Check supported rules below

**Pull request merged**